embeddings/task/lightning_task/sequence_labeling.py

```diff
@@ -33,11 +35,15 @@ def __init__(
     def build_task_model(self) -> None:
         self.model = SequenceLabelingModule(
             model_name_or_path=self.model_name_or_path,
+            num_classes=self.num_classes,
             finetune_last_n_layers=self.finetune_last_n_layers,
             config_kwargs=self.model_config_kwargs,
             task_model_kwargs=self.task_model_kwargs,
         )
 
+    def restore_task_model(self, checkpoint_path: str) -> None:
+        self.model = SequenceLabelingModule.load_from_checkpoint(checkpoint_path)
```

Comment:
`classmethod` shouldn't be more unifed? And maybe just rename it to `load` ?

Reply:
on the second thought, I will remove these methods since we need to initialize the Task to use them so there is not much use for them anyway

Reply:
we would need a clever way to initialize the task without the need for specifying the arguments in the `__init__`

Reply:
I have added one idea in comment above that could handle this issue

Reply:
this was completely refactored in next commits so you can check how it works now